.github/workflows/build.yml

```diff
@@ -0,0 +1,170 @@
+name: Build FreeOrion and run unit tests
+on:
+  pull_request:
+    types:
+      - opened
+      - edited
+      - reopened
+      - synchronize
+  push:
+    branches:
+      - master
+jobs:
+  build-linux:
+    name: Build FreeOrion on Ubuntu 18.04 (Bionic)
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        id: build-linux-install-cache-id
+        with:
+          path: ${{ runner.temp }}/cache-build-linux-install
+          key: build-linux-install-cache-key-v1.0
+      - uses: airvzxf/cache-anything-new-action@v1.0
+        with:
+          script: 'install-build.sh'
+          is_cached: ${{ steps.build-linux-install-cache-id.outputs.cache-hit }}
+          cache: ${{ runner.temp }}/cache-build-linux-install
+          snapshot: '/'
+          exclude: '/boot /data /dev /mnt /proc /run /sys'
+      - name: Prepare ccache timestamp
+        id: build-linux-ccache-timestamp
+        shell: cmake -P {0}
+        run: |
+          string(TIMESTAMP current_date "%Y-%m-%d-%H;%M;%S" UTC)
+          message("::set-output name=timestamp::${current_date}")
+      - name: ccache files
+        uses: actions/cache@v1.1.0
+        with:
+          path: ${{ runner.temp }}/ccache
+          key: build-linux-ccache-${{ steps.build-linux-ccache-timestamp.outputs.timestamp }}
+          restore-keys: |
+            build-linux-ccache-
+      - name: Build
+        env:
+          CACHE_NAME: linux
+          CCACHE_DIR: ${{ runner.temp }}/ccache
+          CMAKE_BUILD_PARALLEL_LEVEL: 2
+        run: |
+          echo "----------------------------------------"
+          ccache -s || true
+          echo "----------------------------------------"
+          mkdir build
+          cd build
+          cmake -DBUILD_TESTING=ON ..
+          cmake --build .
+          echo "----------------------------------------"
+          ccache -s || true
```

Comment:
We have 2 stats call. They are run in different directories, does it affects anything?

Could we move them to separate steps before and after the build step?  There might be many reasons to see the build log and checking cache is only one of them.  In most cases, cache statistics are not interesting for developers.  Also, you could remove echo parts since they will be in different sections.  Moreover, it would be much easier to compare them because build logs are huge.

Steps in UI:
![image](https://user-images.githubusercontent.com/171597/107885179-d1317400-6f09-11eb-823b-4da36696776c.png)

Reply:
Moved to separate step.